### ISSUE TITLE
[Angular] Build fails if the name in the package.json does not match with the name of the project in angular.json

### ISSUE THREAD

**Issue by ThoSap** [quarkiverse/quarkus-quinoa#721]
### Describe the bug

Hi @yatho @melloware,

since Quinoa 2.4.2 a build fails if the `package.json` name does not match the name of the Angular projects `angular.json`.
This is caused by PR #716 in lines
https://github.com/quarkiverse/quarkus-quinoa/blob/2.4.2/deployment/src/main/java/io/quarkiverse/quinoa/deployment/framework/override/AngularFramework.java#L36
and
https://github.com/quarkiverse/quarkus-quinoa/blob/2.4.2/deployment/src/main/java/io/quarkiverse/quinoa/deployment/framework/override/AngularFramework.java#L56-L59

This PR change was really necessary with the new Angular 17+ ESBuild + Vite builder and I appended `/browser` to my custom `quarkus.quinoa.build-dir` configuration option in the past manually with older Quinoa versions.

The build will also fail if the `package.json` name is not set at all, as this code then falls back to `quinoa`
```java
final String applicationName = packageJson.map(p -> p.getString("name")).orElse("quinoa");
```
which will most likely not exist as a project in the `angular.json`.

In the meantime, I aligned the names in my project, which fixes this build issue.

Exception:
```java
> Task :quarkusAppPartsBuild FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':quarkusAppPartsBuild'.
> There was a failure while executing work items
   > A failure occurred while executing io.quarkus.gradle.tasks.worker.BuildWorker
      > io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        	[error]: Build step io.quarkiverse.quinoa.deployment.QuinoaProcessor#processBuild threw an exception: java.lang.NullPointerException: Cannot invoke "jakarta.json.JsonObject.getJsonObject(String)" because the return value of "jakarta.json.JsonObject.getJsonObject(String)" is null
        	at io.quarkiverse.quinoa.deployment.framework.override.AngularFramework$1.getBuilderUse(AngularFramework.java:58)
        	at io.quarkiverse.quinoa.deployment.framework.override.AngularFramework$1.buildDir(AngularFramework.java:38)
        	at io.quarkiverse.quinoa.deployment.QuinoaProcessor.processBuild(QuinoaProcessor.java:175)
        	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:849)
        	at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
        	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
        	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
        	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
        	at java.base/java.lang.Thread.run(Thread.java:1583)
        	at org.jboss.threads.JBossThread.run(JBossThread.java:483)
```

### Quinoa version

2.4.2

### Quarkus version

3.12.3

### Build / Runtime

Angular

### Package Manager

PNPM

### Steps to reproduce the behavior

Quarkus + Quinoa [Angular] + the `package.json` name does not match the `angular.json` project name.

### Expected behavior

If it really is the case that the `package.json` name and `angular.json` project name do not match:

1. Try to use the first key in the `angular.json` project Object as a fallback (or a better detection logic) where the `architect.build.builder` key exists
2. **Log a warning that the name does not match and that it should be fixed by the developer**

**Comment by yatho** [comment]
It is exact if the application is not created in the same time and with the same name than the workspace it will be incorrect. I will change that. Sorry for the inconvenience.

**Comment by ThoSap** [comment]
Hey @yatho, absolutely no problem, it took me like 2 minutes to track down and fix the issue in my build :)
Thank you for your previous PR, it is a nice addition to Quinoa! 😊

**Comment by melloware** [comment]
@yatho after you fix i can push a 2.4.3 release.
